From 9d0913acd6536c7f0f781392557be740f392137c Mon Sep 17 00:00:00 2001
From: Maruthi Srinivas Bayyavarapu <maruthi.srinivas.bayyavarapu@xilinx.com>
Date: Tue, 16 Oct 2018 20:14:24 +0530
Subject: [PATCH 0128/1851] ASoC: xlnx: Set missing/invalid channel count to
 default value in consumer format

commit 239269f7e0ee680249c591bb88c48f6596822ba3 from
https://github.com/Xilinx/linux-xlnx.git

Set channel count to default value 2 when the captured stream has
invalid or missing channel count.

Signed-off-by: Maruthi Srinivas Bayyavarapu <maruthi.srinivas.bayyavarapu@xilinx.com>
Reviewed-by: Vishal Sagar <vishal.sagar@xilinx.com>
Signed-off-by: Michal Simek <michal.simek@xilinx.com>
State: pending
Signed-off-by: Yaliang Wang <Yaliang.Wang@windriver.com>
---
 sound/soc/xilinx/xlnx_formatter_pcm.c | 7 +++++--
 1 file changed, 5 insertions(+), 2 deletions(-)

diff --git a/sound/soc/xilinx/xlnx_formatter_pcm.c b/sound/soc/xilinx/xlnx_formatter_pcm.c
index 585172c73842..48e584b0ec17 100644
--- a/sound/soc/xilinx/xlnx_formatter_pcm.c
+++ b/sound/soc/xilinx/xlnx_formatter_pcm.c
@@ -273,8 +273,11 @@ static int parse_consumer_format(u32 chsts_reg1_val, u32 chsts_reg2_val,
 	       CON_CHANNEL_COUNT_SHIFT;
 	params->channels = val;
 
-	/* if not found, set it to default */
-	if (!params->channels)
+	/*
+	 * if incorrect channel count embedded is less than 2, set it to
+	 * supported default which is 2 in our case
+	 */
+	if (params->channels < 2)
 		params->channels = 2;
 
 	if (chsts_reg2_val & CON_MAX_BITDEPTH_MASK)
-- 
2.31.1

